### PR TITLE
[FEATURE] Add authentication by client certificate

### PR DESCRIPTION
**Pull request by norbertvolf**
The patch implement authentication via client ceriticates in PEM and PFX format. New parameters ca, cert, key, pfx and passphrase added to initialization. The client certificat authentication is not part of "automatic" determination of authentication type.

Topic: #81

### CODE REVIEW

File: doc/INITIALIZE.md [hunk 1]
```diff
@@ -85,7 +181,7 @@ Set cookie via environment variable
 
 ```shell
 export ODATA_COOKIE="JSESSIONID=s:bdzps02ARlShtevVcSWsTLptzhPdAF-y.r2nlOcl38jriMxfIhcvIzyFwS0V9nITPUz8orkAHMic"
-```
+````
```
Comment (jakub-vaclavik-sap) [inline doc/INITIALIZE.md:184]: typo

File: doc/INITIALIZE.md [hunk 2]
```diff
@@ -72,11 +71,108 @@ The odata-library currently support four authentication types.
...
 - cookie - run authentication code externally and just pass authentication cookies to odata-library
+- cert - pass client certificate in PEM format
 
 If authentication type is not specified odata-library tries to use first three authentication
 types automatically.
 
-## Authenticate to service outside of Service
+### Client certificate authentication
+
+The client certificate settings are passed to https.Agent which is used
```
Comment (jakub-vaclavik-sap) [inline doc/INITIALIZE.md:81]: missing space after dot ('https.Agent')

File: doc/INITIALIZE.md (hunk 1 above)
Comment (norbertvolf) [reply doc/INITIALIZE.md]: Fixed

File: doc/INITIALIZE.md (hunk 2 above)
Comment (norbertvolf) [reply doc/INITIALIZE.md:81]: Hi @jakub-vaclavik-sap ,

I am not understand. https.Agent is class from NodeJS.

https://nodejs.org/api/https.html#class-httpsagent

I have added the URL to the documentation.

Comment (jakub-vaclavik-sap) [reply doc/INITIALIZE.md]: My bad, you are right.

File: lib/agent/settings.js [hunk 3]
```diff
@@ -261,4 +249,167 @@ parseSettings._.checkCookieSettings = function (connectionSettings) {
...
+        source: def.SOURCE === "ENV" ? processEnv : connectionSettings,
+      },
+      def
+    )
+  )
+    .sort((defA, defB) => defA.ORDER - defB.ORDER)
+    .find((def) => def.MANDATORY_KEYS.some((key) => _.has(def.source, key)));
+};
+
+parseSettings._.checkTLSDefinition = function (
```
Comment (jakub-vaclavik-sap) [inline lib/agent/settings.js:320]: If I'am not missing something, this function is not used, it is redefined on ln 349 and the second definition is used.

Comment (norbertvolf) [reply lib/agent/settings.js]: You are right. It was my copy&paste issue.

**Review by michalkopulety — APPROVED** [review]

**Pull request merged**